nemo/collections/asr/parts/submodules/multi_head_attention.py

```diff
@@ -219,19 +216,24 @@ class PositionalEncoding(torch.nn.Module):
         dropout_rate_emb (float): dropout rate for the positional embeddings
     """
 
-    def __init__(self, d_model, dropout_rate, max_len=5000, xscale=None, dropout_rate_emb=0.0):
+    def __init__(self, d_model, dropout_rate, max_len=5000, xscale=None, dropout_rate_emb=0.0, positions=None):
         """Construct an PositionalEncoding object."""
         super(PositionalEncoding, self).__init__()
         self.d_model = d_model
         self.xscale = xscale
         self.dropout = torch.nn.Dropout(p=dropout_rate)
-        self.extend_pe(length=max_len)
+        if positions is None:
+            positions = torch.arange(0, max_len, dtype=torch.float32)
+        self.create_pe(positions=positions)
+        self.center_pos = torch.tensor(self.pe.size(1) // 2 + 1, dtype=torch.int32)
```

Comment:
self.center_pos is not used in PositionalEncoding. It is a concept which is needed for RelPositionalEncoding. We may need to move it to the init of RelPositionalEncoding or calculate it on the fly.